### PR TITLE
Bump plugins to 2.0 and support build.version_qualifier

### PR DESCRIPTION
**Pull request by joshuali925**
### Description
- bump frontend and backend to 2.0
- update gradle and kotlin
- update tests
- drop java 8 support
- support version qualifier

TODO
- cypress test pending on SQL/PPL 2.0
- frontend version qualifier

### Issues Resolved
partially #577, need instructions on how to support version qualifier for frontend plugin

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: opensearch-observability/src/test/kotlin/org/opensearch/observability/PluginRestTestCase.kt [hunk 1]
```diff
@@ -263,7 +263,7 @@ abstract class PluginRestTestCase : OpenSearchRestTestCase() {
                     false
                 )
                 proxy.getExecutionData(false)?.let {
-                    val path = Paths.get("$jacocoBuildPath/integTest.exec")
```
Comment (vamsi-amazon) [inline opensearch-observability/src/test/kotlin/org/opensearch/observability/PluginRestTestCase.kt:266]: Is this a part of JDK-8 removal?

File: opensearch-observability/src/main/kotlin/org/opensearch/observability/index/ObservabilityIndex.kt [hunk 2]
```diff
@@ -122,7 +122,6 @@ internal object ObservabilityIndex {
         val classLoader = ObservabilityIndex::class.java.classLoader
         val indexMappingSource = classLoader.getResource(OBSERVABILITY_MAPPING_FILE_NAME)?.readText()!!
         val request = PutMappingRequest(INDEX_NAME)
-            .type("_doc")
```
Comment (vamsi-amazon) [inline opensearch-observability/src/main/kotlin/org/opensearch/observability/index/ObservabilityIndex.kt:125]: In response also, there wont be _doc. I am hoping this is handled.

File: opensearch-observability/src/test/kotlin/org/opensearch/observability/PluginRestTestCase.kt (hunk 1 above)
Comment (joshuali925) [reply opensearch-observability/src/test/kotlin/org/opensearch/observability/PluginRestTestCase.kt:266]: yes

**Review by kavithacm — APPROVED** [review]

**Comment by joshuali925** [comment]
there are still pending issues i'll send a follow up PR to fix, merging this in first

**Pull request merged**